docs/docs/code/python/http-requests/README.md

```diff
@@ -39,24 +83,215 @@ def handler(pd: "pipedream"):
+
+Here's an example showing an API key passed in an HTTP header:
+
+```python
+import requests
+
+def handler(pd: "pipedream"):
+  url = "https://jsonplaceholder.typicode.com/posts"
+  headers = {"X-API-KEY": "123"} # API KEY
+  data = {"name": "Luke"}
+
+  # Make an HTTP POST request using requests
+  r = requests.post(url, headers=headers, json=data)
+```
 
-You can also send files within a step.
+## Sending files
 
 An example of sending a previously stored file in the workflow's `/tmp` directory:
 
 ```python
-# Retrieving a previously saved file from workflow storage
-files = {"image": open("/tmp/python-logo.png", "rb")}
+import requests
+
+def handler(pd: "pipedream"):
+  # Retrieving a previously saved file from workflow storage
+  files = {"image": open("/tmp/python-logo.png", "rb")}
 
-r = requests.post(url="https://api.imgur.com/3/image", files=files)
+  r = requests.post(url="https://api.imgur.com/3/image", files=files)
 ```
+
+## Downloading a file to the `/tmp` directory
+
+This example shows you how to download a file to a file in [the `/tmp` directory](/code/python/working-with-files/). This can be especially helpful for downloading large files: it streams the file to disk, minimizing the memory the workflow uses when downloading the file.
+
+```python
+import requests
+
+def handler(pd: "pipedream"):
+  # Download the webpage HTML file to /tmp
+  with requests.get("https://example.com", stream=True) as response:
+      # Check if the request was successful
+      response.raise_for_status()
+
+      # Open the new file /tmp/file.html in binary write mode
+      with open("/tmp/file.html", "wb") as file:
+          for chunk in response.iter_content(chunk_size=8192):
+              # Write the chunk to file
+              file.write(chunk)
+```
+
+## Uploading a file from the `/tmp` directory
+
+This example shows you how to make a `multipart/form-data` request with a file as a form part. You can store and read any files from [the `/tmp` directory](/code/python/working-with-files/#the-tmp-directory).
+
+This can be especially helpful for uploading large files: it streams the file from disk, minimizing the memory the workflow uses when uploading the file.
+
+```python
+import requests
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+def handler(pd: "pipedream"):
+  m = MultipartEncoder(fields={
+    'file': ('filename', open('/tmp/file.pdf', 'rb'), 'application/pdf')
+    })
+
+  r = requests.post("https://example.com", data=m,
+                  headers={'Content-Type': m.content_type})
+```
+
+## IP addresses for HTTP requests made from Pipedream workflows
+
+By default, [HTTP requests made from Pipedream can come from a large range of IP addresses](/privacy-and-security/#hosting-details). **If you need to restrict the IP addresses HTTP requests come from, you can [Use a Pipedream VPC](/workflows/vpc/) to route all outbound HTTP requests through a single IP address.**
```

Comment:
I didn't include using `pd.send.http` as an option because it didn't work for me in testing. E.g., from Node.js:

https://github.com/PipedreamHQ/pipedream/blob/284df1945cd4a94c30ed929794f9a23eb44e5bf9/docs/docs/code/nodejs/http-requests/README.md#L496